core/scripts/chaincli/handler/migrate_cron.go

```diff
@@ -0,0 +1,228 @@
+package handler
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	cron_factory "github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/cron_upkeep_factory_wrapper"
+	proxy "github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/permissioned_forward_proxy_wrapper"
+	registrar "github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/upkeep_registration_requests_wrapper"
+	helpers "github.com/smartcontractkit/chainlink/core/scripts/common"
+)
+
+func (k *Keeper) MigrateCron(ctx context.Context, inputFilePath string) {
+	proxyAddr := common.HexToAddress(k.cfg.ProxyAddr)
+	proxyAbi, err := abi.JSON(strings.NewReader(proxy.PermissionedForwardProxyABI))
+	if err != nil {
+		log.Fatalln("Error generating proxy ABI", err)
+	}
+
+	// Throw error if output file already exists
+	outputFile, err := os.OpenFile("migrate_cron_output.csv", os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
```

Comment:
are we missing an explicit close for this?

Reply:
(y) yep, i think i missed it